core.c

```diff
@@ -496,6 +496,25 @@ int readbyte(FILE * f)
     return strtol(data, NULL, 16);
 }
 
+int load_raw(struct em8051 *aCPU, char *aFilename)
+{
+    FILE *f;
+    if (aFilename == 0 || aFilename[0] == 0)
+        return -1;
+    f = fopen(aFilename, "r");
+    if (!f) return -1;
+
+    int address = 0;
+    while (!feof(f))
```

Comment:
This loop will happily try to write megabytes of stuff into memory.. it would probably be a good idea to limit read to the actual target buffer size.